File: app/src/main/java/com/bq/zowi/models/commands/LeftRightCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;

/* JADX INFO: loaded from: classes.dex */
public class LeftRightCommand extends MovementCommand {
    private static final Command.Action[] LEFT_RIGHT_ACTIONS = {Command.Action.TURN, Command.Action.MOONWALKER, Command.Action.CRUSAITO, Command.Action.BEND, Command.Action.SHAKE_LEG};
    private static final Command.Direction[] LEFT_RIGHT_DIRECTIONS = {Command.Direction.LEFT, Command.Direction.RIGHT};

    public LeftRightCommand(Command.Action action, Command.Direction direction, long duration) {
        setAction(action);
        setDirection(direction);
        setDuration(duration);
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action[] getAllowedActions() {
        return LEFT_RIGHT_ACTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Direction[] getAllowedDirections() {
        return LEFT_RIGHT_DIRECTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command copy() {
        return new LeftRightCommand(getAction(), getDirection(), getDuration().longValue());
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/LeftRightCommand.kt
package com.bq.zowi.models.commands

class LeftRightCommand(
    action: Command.Action,
    direction: Command.Direction,
    duration: Long
) : MovementCommand() {

    init {
        setAction(action)
        setDirection(direction)
        setDuration(duration)
    }

    override fun getAllowedActions(): Array<Command.Action> = LEFT_RIGHT_ACTIONS
    override fun getAllowedDirections(): Array<Command.Direction> = LEFT_RIGHT_DIRECTIONS
    override fun copy(): Command = LeftRightCommand(getAction(), getDirection()!!, getDuration()!!)

    companion object {
        private val LEFT_RIGHT_ACTIONS = arrayOf(
            Command.Action.TURN, Command.Action.MOONWALKER, Command.Action.CRUSAITO,
            Command.Action.BEND, Command.Action.SHAKE_LEG
        )
        private val LEFT_RIGHT_DIRECTIONS = arrayOf(Command.Direction.LEFT, Command.Direction.RIGHT)
    }
}
